src/Symfony/Component/Notifier/Bridge/AmazonSns/AmazonSnsTransportFactory.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\AmazonSns;
+
+use AsyncAws\Sns\SnsClient;
+use Symfony\Component\Notifier\Exception\UnsupportedSchemeException;
+use Symfony\Component\Notifier\Transport\AbstractTransportFactory;
+use Symfony\Component\Notifier\Transport\Dsn;
+use Symfony\Component\Notifier\Transport\TransportInterface;
+
+/**
+ * @author Adrien Chinour <github@chinour.fr>
+ *
+ * @experimental in 5.3
+ */
+final class AmazonSnsTransportFactory extends AbstractTransportFactory
+{
+    private const DSN_SCHEME = 'sns';
+
+    public function create(Dsn $dsn): TransportInterface
+    {
+        $scheme = $dsn->getScheme();
+
+        if (self::DSN_SCHEME === $scheme) {
+            $options = [
+                    'region' => $dsn->getOption('region') ?: 'eu-west-1',
+                    'profile' => $dsn->getOption('profile'),
+                    'accessKeyId' => $dsn->getUser(),
+                    'accessKeySecret' => $dsn->getPassword(),
+                ] + (
+                'default' === $dsn->getHost() ? [] : ['endpoint' => 'https://'.$dsn->getHost().($dsn->getPort() ? ':'.$dsn->getPort() : '')]
+                );
+
+            return new AmazonSnsTransport(new SnsClient($options, null, $this->client), $this->client, $this->dispatcher);
+        }
+
+        throw new UnsupportedSchemeException($dsn, self::DSN_SCHEME, $this->getSupportedSchemes());
+    }
```

Comment:
Please pull up the scheme check like here, to first check if the scheme is supported and afterwards for missing options. Thanks

https://github.com/symfony/symfony/blob/7d846d3e82f4e3dabec7d4b5300f50a1dd9ed7d6/src/Symfony/Component/Notifier/Bridge/Sinch/SinchTransportFactory.php#L27-L47